### PR TITLE
Add operators to futures

### PR DESCRIPTION
**Pull request by MrLotU**
Signed-off-by: MrLotU <j.koopman@jarict.nl>

### CODE REVIEW

File: Tests/NIOKitTests/XCTestManifests.swift [hunk 1]
```diff
@@ -4,6 +4,7 @@ import XCTest
 public func allTests() -> [XCTestCaseEntry] {
     return [
         testCase(NIOKitTests.allTests),
+        testCae(FutureOperatorsTests.allTests)
```
Comment (tanner0101) [inline Tests/NIOKitTests/XCTestManifests.swift]: typo here

File: Tests/NIOKitTests/Extensions/FutureOperatorsTests.swift [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        future1 |= future2
+        XCTAssertEqual(try future1.wait(), 13)
+    }
+    
+    func testNOT() throws {
+        let future1: EventLoopFuture<UInt8> = eventLoop.newSucceededFuture(result: 0b00001111)
+        XCTAssertEqual(try ~future1.wait(), 0b11110000)
+    }
+    
+    static var allTests = [
```
Comment (tanner0101) [inline Tests/NIOKitTests/Extensions/FutureOperatorsTests.swift]: I think these need to be tuples with references to the methods

File: Sources/NIOKit/Extensions/FutureOperators.swift [hunk 3]
```diff
@@ -0,0 +1,290 @@
+//
+//  FutureOperators.swift
+//  NIOKit
+//
+//  Created by Jari (LotU) on 06/11/2018.
```
Comment (tanner0101) [inline Sources/NIOKit/Extensions/FutureOperators.swift]: rm Xcode header

File: Sources/NIOKit/Extensions/FutureOperators.swift [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+//
+
+import NIO
+
+// MARK: - +
```
Comment (tanner0101) [inline Sources/NIOKit/Extensions/FutureOperators.swift]: what's the `-` mean in all these MARKs ?

Comment (MrLotU) [reply Sources/NIOKit/Extensions/FutureOperators.swift]: It's to make seperators in Xcode

File: Sources/NIOKit/Extensions/FutureOperators.swift (hunk 3 above)
Comment (MrLotU) [reply Sources/NIOKit/Extensions/FutureOperators.swift]: 👍

**Comment by MrLotU** [comment]
@tanner0101 @calebkleveter Tests are working! Can I have a final review? :)

**Comment by MrLotU** [comment]
screenshot 2018-11-08 at 22 00 34
@tanner0101 The reason I used `// MARK: - <things>`

File: Tests/NIOKitTests/Extensions/FutureOperatorsTests.swift [hunk 5]
```diff
@@ -0,0 +1,164 @@
+//
+//  FutureOperatorsTests.swift
+//
+//  Created by Jari (LotU) on 06/11/2018.
+//
```
Comment (calebkleveter) [inline Tests/NIOKitTests/Extensions/FutureOperatorsTests.swift]: This Xcode header should be removed also.

Comment (MrLotU) [reply Tests/NIOKitTests/Extensions/FutureOperatorsTests.swift]: Out of curiosity, why do you not want those headers there?

Comment (calebkleveter) [reply Tests/NIOKitTests/Extensions/FutureOperatorsTests.swift]: Tenner said to remove them ¯\_(ツ)_/¯ https://github.com/vapor-community/nio-kit/pull/3#discussion_r232040823

Comment (MrLotU) [reply Tests/NIOKitTests/Extensions/FutureOperatorsTests.swift]: What I got from that is just the ```swift
// NIOKit
```
part which indicates the XCode module, but that might be me

Comment (MrLotU) [reply Tests/NIOKitTests/Extensions/FutureOperatorsTests.swift]: I'll ask Tanner once he's awake :)

Comment (tanner0101) [reply Tests/NIOKitTests/Extensions/FutureOperatorsTests.swift]: Generally, I don't include any headers in Vapor files. The git repo should contain any info you'd want to include in the headers (such as author, license, etc). Only special cases, like generated files, deserve a header IMO.

Comment (MrLotU) [reply Tests/NIOKitTests/Extensions/FutureOperatorsTests.swift]: 👍 I'll remove all of them

**Comment by MrLotU** [comment]
🎉

**Pull request merged**